agent/reconciler/host_reconciler.go

```diff
@@ -151,12 +155,28 @@ func (r *HostReconciler) SetupWithManager(ctx context.Context, mgr manager.Manag
 		Complete(r)
 }
 
-// cleanup kubeadm dir to remove any stale config on the host
-func (r *HostReconciler) kubeadmDirCleanup(ctx context.Context) error {
+// cleanup /run/kubeadm, /etc/cni/net.d dirs to remove any stale config on the host
+func (r *HostReconciler) cleank8sdirectories(ctx context.Context) error {
 	logger := ctrl.LoggerFrom(ctx)
-	logger.Info("cleaning up kubeadm directory")
-	const kubeadmDir = "/run/kubeadm"
-	return os.RemoveAll(kubeadmDir)
+
+	dirs := []string{
+		"/run/kubeadm/*",
+		"/etc/cni/net.d/*",
+	}
+
+	errList := []error{}
+	for _, dir := range dirs {
+		logger.Info(fmt.Sprintf("cleaning up directory %s", dir))
+		if err := common.RemoveGlob(dir); err != nil {
+			logger.Error(err, fmt.Sprintf("failed to clean up directory %s", dir))
+			errList = append(errList, err)
+		}
+	}
+
+	if len(errList) > 0 {
+		return errors.New("not all k8s directories are cleaned up")
```

Comment:
Can we add more info?
```suggestion
		return errors.New("not all k8s directories are cleaned up\n cleanup /run/kubeadm and /etc/sni/net.d to proceed")
```

Reply:
It's better not have "\n" and so specified message in an error object. And I also have[ error messag](https://github.com/huchen2021/cluster-api-provider-bringyourownhost/blob/adff684ef2df96297a2d90cada7cd7a6e79e9deb/agent/reconciler/host_reconciler.go#L171)e to indicated which directory clean failed.

Reply:
I guess we are anyways logging the errors, so here this could be a generic one as it will not be consumed by some downstream systems. wdyt?

Reply:
sounds good.